src/utils/validations/user.validations.ts

```diff
@@ -11,4 +11,27 @@ const userLoginValidation = z
       .regex(passwordStrength, "You are using wrong password please try again")
   })
   .strict();
-export { userLoginValidation };
+const userUpdatePassValidation = z.object({
+  oldPassword: z
+    .string()
+    .min(8, "password of length less than 8")
+    .regex(
+      passwordStrength,
+      "You are using wrong old password please try again"
+    ),
+  newPassword: z
+    .string()
+    .min(8, "password of length less than 8")
+    .regex(
+      passwordStrength,
+      "You are using wrong new password please try again"
+    ),
+  confirmPassword: z
+    .string()
+    .min(8, "password of length less than 8")
```

Comment:
At here you can instead check if this password equals to the new password provided.
You can refer to this [example](https://stackoverflow.com/questions/75572170/zod-validation-based-on-another-field) or [this](https://stackoverflow.com/questions/75572170/zod-validation-based-on-another-field).